In [1]:
import pandas as pd
import numpy as np
import statistics
import scipy.stats as stats

1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   Region A: [10, 15, 12, 8, 14]
   Region B: [18, 20, 16, 22, 25]
   Calculate the mean sales for each region.


In [2]:
data={'Region A': [10, 15, 12, 8, 14], 'Region B':[18, 20, 16, 22, 25]}
df=pd.DataFrame(data)
print(df.mean())

Region A    11.8
Region B    20.2
dtype: float64


In [3]:
data={'Rating': [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]}
df=pd.DataFrame(data)
print(df.mode())

   Rating
0       4
1       5


In [4]:
import statistics
survey_responses = [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
mode = statistics.mode(survey_responses)
print(mode)

4


3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
   Department A: [5000, 6000, 5500, 7000]
   Department B: [4500, 5500, 5800, 6000, 5200]
   Calculate the median salary for each department.


In [5]:
department_a_salaries = [5000, 6000, 5500, 7000]
department_b_salaries = [4500, 5500, 5800, 6000, 5200]

median_a = np.median(department_a_salaries)
median_b = np.median(department_b_salaries)

print("Median salary for Department A:", median_a)
print("Median salary for Department B:", median_b)

Median salary for Department A: 5750.0
Median salary for Department B: 5500.0


4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
   [25.5, 24.8, 26.1, 25.3, 24.9]
   Calculate the range of the stock prices.


In [6]:
stock_prices = pd.Series([25.5, 24.8, 26.1, 25.3, 24.9])
price_range = np.ptp(stock_prices)

print("Range of stock prices: %0.1f" %price_range)

Range of stock prices: 1.3


5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
   Group A: [85, 90, 92, 88, 91]
   Group B: [82, 88, 90, 86, 87]
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups.


In [7]:
group_a_scores = [85, 90, 92, 88, 91]
group_b_scores = [82, 88, 90, 86, 87]

t_statistic, p_value = stats.ttest_ind(group_a_scores, group_b_scores)

print("T-statistic:", t_statistic)
print("P-value:", p_value)

T-statistic: 1.4312528946642733
P-value: 0.19023970239078333


In this case, the t-statistic is approximately 1.431 and the p-value is approximately 0.190. Since the p-value is greater than 0.05, we do not have enough evidence to reject the null hypothesis and conclude that there is a significant difference in the mean scores between the two groups.

6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
   Sales (in thousands): [25, 30, 28, 20, 26]
   Calculate the correlation coefficient between advertising expenditure and sales.


In [8]:
advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

correlation_coefficient = np.corrcoef(advertising_expenditure, sales)[0]

print("Correlation coefficient:", correlation_coefficient)

Correlation coefficient: [1.         0.87575114]


The resulting output is the correlation coefficient, which is approximately 0.875. The correlation coefficient ranges from -1 to 1, where values closer to 1 indicate a strong positive correlation.

7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   [160, 170, 165, 155, 175, 180, 170]
   Calculate the standard deviation of the heights.


In [9]:
heights = [160, 170, 165, 155, 175, 180, 170]

standard_deviation = np.std(heights)

print("Standard deviation of heights:", standard_deviation)


Standard deviation of heights: 7.953949089757174


8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.


In [10]:
from sklearn.linear_model import LinearRegression

employee_tenure = np.array([2, 3, 5, 4, 6, 2, 4]).reshape(-1, 1)
job_satisfaction = np.array([7, 8, 6, 9, 5, 7, 6])
model = LinearRegression()
model.fit(employee_tenure, job_satisfaction)

print("Intercept:", model.intercept_)
print("Coefficient for employee tenure:", model.coef_[0])


Intercept: 8.595744680851062
Coefficient for employee tenure: -0.4680851063829787


9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
   Medication A: [10, 12, 14, 11, 13]
   Medication B: [15, 17, 16, 14, 18]
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.


In [11]:
medication_a_recovery_times = [10, 12, 14, 11, 13]
medication_b_recovery_times = [15, 17, 16, 14, 18]

f_statistic, p_value = stats.f_oneway(medication_a_recovery_times, medication_b_recovery_times)

print("F-statistic:", f_statistic)
print("P-value:", p_value)

F-statistic: 16.0
P-value: 0.003949772803445326


10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is

 as follows:
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    Calculate the 75th percentile of the feedback ratings.


In [12]:
feedback_ratings = [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

percentile_75 = np.percentile(feedback_ratings, 75)

print("75th percentile of feedback ratings:", percentile_75)


75th percentile of feedback ratings: 8.75


11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.


In [13]:
weights = [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
population_mean = 10

t_statistic, p_value = stats.ttest_1samp(weights, population_mean)

print("T-statistic:", t_statistic)
print("P-value:", p_value)

T-statistic: 1.5126584522688367
P-value: 0.19077595151110102


12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
    Design A: [100, 120, 110, 90, 95]
    Design B: [80, 85, 90, 95, 100]
    Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs.


In [14]:
clicks_design_a = [100, 120, 110, 90, 95]
clicks_design_b = [80, 85, 90, 95, 100]

contingency_table = np.array([clicks_design_a, clicks_design_b])

chi2_statistic, p_value, _, _ = stats.chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2_statistic)
print("P-value:", p_value)

Chi-square statistic: 6.110658166925435
P-value: 0.19103526314060293


13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    Calculate the 95% confidence interval for the population mean satisfaction score.


In [15]:
satisfaction_scores = [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]

confidence_level = 0.95
sample_mean = np.mean(satisfaction_scores)
sample_std = np.std(satisfaction_scores, ddof=1)
sample_size = len(satisfaction_scores)

margin_of_error = stats.t.ppf(1 - (1 - confidence_level) / 2, df=sample_size - 1) * (sample_std / np.sqrt(sample_size))

confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

print("95% Confidence Interval:", confidence_interval)


95% Confidence Interval: (7.043561120599888, 8.756438879400113)


14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    Perform a simple linear regression to predict performance based on temperature.


In [16]:
temperature = np.array([20, 22, 23, 19, 21]).reshape(-1, 1)
performance = np.array([8, 7, 9, 6, 8])

model = LinearRegression()
model.fit(temperature, performance)

# Print the regression coefficients
print("Intercept:", model.intercept_)
print("Coefficient for temperature:", model.coef_[0])


Intercept: -2.8999999999999986
Coefficient for temperature: 0.4999999999999999


15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
    Group A: [4, 3, 5, 2, 4]
    Group B: [3, 2, 4, 3, 3]
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.


In [17]:
import scipy.stats as stats

group_a_preferences = [4, 3, 5, 2, 4]
group_b_preferences = [3, 2, 4, 3, 3]

statistic, p_value = stats.mannwhitneyu(group_a_preferences, group_b_preferences)

print("Mann-Whitney U statistic:", statistic)
print("P-value:", p_value)

Mann-Whitney U statistic: 17.0
P-value: 0.380836480306712


16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    Calculate the interquartile range (IQR) of the ages.


In [18]:
ages = [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

q1 = np.percentile(ages, 25)
q3 = np.percentile(ages, 75)

iqr = q3 - q1

print("Interquartile Range (IQR) of ages:", iqr)


Interquartile Range (IQR) of ages: 22.5


17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]
    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]
    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]
    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.


In [19]:
algorithm_a_scores = [0.85, 0.80, 0.82, 0.87, 0.83]
algorithm_b_scores = [0.78, 0.82, 0.84, 0.80, 0.79]
algorithm_c_scores = [0.90, 0.88, 0.89, 0.86, 0.87]

statistic, p_value = stats.kruskal(algorithm_a_scores, algorithm_b_scores, algorithm_c_scores)

print("Kruskal-Wallis statistic:", statistic)
print("P-value:", p_value)

Kruskal-Wallis statistic: 9.696947935368053
P-value: 0.007840333026249539


18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
    Price (in dollars): [10, 15, 12, 8, 14]
    Sales: [100, 80, 90, 110, 95]
    Perform a simple linear regression to predict

 sales based on price.

In [20]:
import scipy.stats as stats

algorithm_a_scores = [0.85, 0.80, 0.82, 0.87, 0.83]
algorithm_b_scores = [0.78, 0.82, 0.84, 0.80, 0.79]
algorithm_c_scores = [0.90, 0.88, 0.89, 0.86, 0.87]

statistic, p_value = stats.kruskal(algorithm_a_scores, algorithm_b_scores, algorithm_c_scores)

print("Kruskal-Wallis statistic:", statistic)
print("P-value:", p_value)

Kruskal-Wallis statistic: 9.696947935368053
P-value: 0.007840333026249539


19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    Calculate the standard error of the mean satisfaction score.


In [21]:
satisfaction_scores = [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

standard_error = stats.sem(satisfaction_scores)

print("Standard Error of the Mean:", standard_error)


Standard Error of the Mean: 0.30550504633038933


20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    Sales (in thousands): [25, 30, 28, 20, 26]
    Perform a multiple regression analysis to predict sales based on advertising expenditure.


In [22]:
advertising_expenditure = np.array([10, 15, 12, 8, 14]).reshape(-1, 1)
sales = np.array([25, 30, 28, 20, 26])

model = LinearRegression()
model.fit(advertising_expenditure, sales)

print("Intercept:", model.intercept_)
print("Coefficient for advertising expenditure:", model.coef_[0])


Intercept: 12.20121951219512
Coefficient for advertising expenditure: 1.152439024390244
